vscripts/confogl.nut

```diff
@@ -2,65 +2,308 @@
+	      local fn = pluginScopes[idx].DirectorOptions.AllowWeaponSpawn;
+	      if ( typeof (fn) == "function" ) {
+		    local res = fn(classname)
+		    // If any plugin returns false, we return false.
+		    if(res == false) {
+		      return false;
+		    }
+		  }
+	    }
+	  }
+    }
+	return retVal;
   }
   // string ConvertWeaponSpawn(string classname)
   // Converts a weapon spawn of given classname to another, used by several mutations.
   function ConvertWeaponSpawn(classname) {
     Msg(format("DirectorOptions.ConvertWeaponSpawn(%s)\n", classname));
+	// Default to no conversion.
+    local retVal = 0;
+	for ( local idx = pluginScopes.len() - 1; idx >= 0; --idx ) {
+      local nextPlugin = pluginScopes[idx];
+	  if ("DirectorOptions" in pluginScopes[idx]) {
+	    if ("ConvertWeaponSpawn" in pluginScopes[idx].DirectorOptions) {
+          local fn = pluginScopes[idx].DirectorOptions.ConvertWeaponSpawn;
+          if ( typeof (fn) == "function" ) {
+            local res = fn(classname)
+		    // If any plugin overrides this function, we return its return value.
+		    return res;
+          }
+		}
+	  }
+    }
+	return retVal;
   }
   // int ConvertZombieClass(infectedClass)
   // Converts one spawn into another, used by the Taaannnk!! mutation (mutation19.nut).
-  function ConvertZombieClass(infectedclass) {
-    Msg(format("DirectorOptions.ConvertWeaponSpawn(%d)\n", infectedclass));
+  function ConvertZombieClass(infectedClass) {
+    Msg(format("DirectorOptions.ConvertZombieClass(%d)\n", infectedClass));
+	// Default to no conversion.
+    local retVal = 0;
+	for ( local idx = pluginScopes.len() - 1; idx >= 0; --idx ) {
+      local nextPlugin = pluginScopes[idx];
+	  if ("DirectorOptions" in pluginScopes[idx]) {
+	    if ("ConvertZombieClass" in pluginScopes[idx].DirectorOptions) {
+          local fn = pluginScopes[idx].DirectorOptions.ConvertZombieClass;
+          if ( typeof (fn) == "function" ) {
+		    local res = fn(infectedClass);
+			// If any plugin overrides this function, we return its return value.
+            return res;
+          }
+	    }
+	  }
+    }
+	return retVal;
   }
   // string GetDefaultItem(index)
   // Repeatedly called with incrementing indices. 
   // Return a string of a weapon name to make it a default item for survivors, or 0 to end the iteration.
   function GetDefaultItem(index) {
     Msg(format("DirectorOptions.GetDefaultItem(%d)\n", index));
+	// Default to ending iteration.
+    local retVal = 0;
+	for ( local idx = pluginScopes.len() - 1; idx >= 0; --idx ) {
+      local nextPlugin = pluginScopes[idx];
+	  if ("DirectorOptions" in pluginScopes[idx]) {
+	    if ("GetDefaultItem" in pluginScopes[idx].DirectorOptions) {
+          local fn = pluginScopes[idx].DirectorOptions.GetDefaultItem;
+          if ( typeof (fn) == "function" ) {
+            local res = fn(index)
+		    // If any plugin overrides this function, we return its return value.
```

Comment:
I kind of think for this function, we should maybe mimic the call order of the builtin hook.

e.g. we call each hook function with 0, 1, 2, 3, 4, ... until they return `0`, and then concatenate the results & return them in the main hook.

Not a super high priority, I'd be happy to do that implementation.

Reply:
That sounds like the sensible solution to me. Returning null or simply returning with no value specified both seem to just end iteration like returning 0 (which removes our default starting single pistol).